packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -56,6 +56,91 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
   onSelect?.(selectedData, newState);
 };
 
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance: TableInstance<T>,
+) => {
+  const resetSelection = () => {
+    instance.rows.forEach((r) => r.toggleRowSelected(false));
+  };
+
+  const getGreaterId = (id: string) => {
+    const greaterId = id.split('.');
+    greaterId[greaterId.length - 1] = `${
+      parseInt(greaterId[greaterId.length - 1]) + 1
+    }`;
+    return greaterId.join('.');
+  };
+
+  const getParentId = (id: string) => {
+    const lastDotIndex = id.lastIndexOf('.');
+    if (lastDotIndex < 0) {
+      return null;
+    }
+    return id.substring(0, lastDotIndex);
+  };
+
+  // Returns true if endId is a parent of currentId
+  // E.g. currentId = 1.4.2.3, endId = 1.4  -> true
+  // E.g. currentId = 1.2.5, endId = 1.3    -> false
+  const isAncestorParent = (currentId: string, endId: string) => {
+    return endId.indexOf(currentId) === 0;
+  };
+
+  // Returns the next row as if it was just a non subrows list
+  const getNextRow = (currentRow: Row<T>, startId: string, endId: string) => {
+    let nextRow;
+    if (
+      (currentRow.subRows ?? []).length > 0 &&
+      isAncestorParent(currentRow.id, endId)
+    ) {
+      // child
+      nextRow = currentRow.subRows[0];
+    } else {
+      // next sibling
+      nextRow = instance.rowsById[getGreaterId(currentRow.id)];
+
+      // If the next sibling doesn't exist, go to the next child at a one upper depth
+      let parentId = getParentId(currentRow.id);
+      while (nextRow == null && parentId != null) {
+        nextRow = instance.rowsById[getGreaterId(parentId)];
+        parentId = getParentId(parentId);
+      }
```

Comment:
On line 106, should I use `!= null` or `!== null`? Is the `!=` version better since it covers more nully/falsy cases? What is the difference?

Reply:
>  Is the `!=` version better since it covers more nully/falsy cases?

That's right